File: src/front/constants/ticketEnums.js
// ============================================
// ticketEnums.js - Fuente única de verdad Frontend
// ============================================

/**
 * Estados del Ticket (formato frontend con guion bajo)
 * Backend usa espacios, Frontend usa guiones bajos
 */
export const TICKET_STATES = {
    CREADO: 'creado',
    EN_ESPERA: 'en_espera',
    EN_PROCESO: 'en_proceso',
    SOLUCIONADO: 'solucionado',
    CERRADO: 'cerrado',
    REABIERTO: 'reabierto'
};

/**
 * Eventos WebSocket del sistema
 */
export const SYSTEM_EVENTS = {
    TICKET_CREATED: 'ticket_created',
    TICKET_ASIGNADO: 'ticket_asignado',
    TICKET_INICIADO: 'ticket_iniciado',
    TICKET_SOLUCIONADO: 'ticket_solucionado',
    TICKET_CERRADO: 'ticket_cerrado',
    TICKET_REABIERTO: 'ticket_reabierto',
    TICKET_ESCALADO: 'ticket_escalado',
    SOLICITUD_REAPERTURA: 'solicitud_reapertura',
    TICKET_ELIMINADO: 'ticket_eliminado',
    ASIGNACION_ELIMINADA: 'asignacion_eliminada'
};

/**
 * Campos especiales del ticket
 */
export const TICKET_PROPS = {
    SOLICITUD_PENDIENTE: 'tiene_solicitud_reapertura_pendiente'
};

/**
 * Mapeo: Evento WebSocket → Estado resultante
 */
export const EVENT_TO_STATE = {
    [SYSTEM_EVENTS.TICKET_CREATED]: TICKET_STATES.CREADO,
    [SYSTEM_EVENTS.TICKET_ASIGNADO]: TICKET_STATES.EN_ESPERA,
    [SYSTEM_EVENTS.TICKET_INICIADO]: TICKET_STATES.EN_PROCESO,
    [SYSTEM_EVENTS.TICKET_SOLUCIONADO]: TICKET_STATES.SOLUCIONADO,
    [SYSTEM_EVENTS.TICKET_CERRADO]: TICKET_STATES.CERRADO,
    [SYSTEM_EVENTS.TICKET_REABIERTO]: TICKET_STATES.REABIERTO,
    [SYSTEM_EVENTS.ASIGNACION_ELIMINADA]: TICKET_STATES.CREADO
};

/**
 * Lista de estados válidos (para validación)
 */
export const VALID_STATES = Object.values(TICKET_STATES);

/**
 * Lista de eventos válidos (para validación)
 */
export const VALID_EVENTS = Object.values(SYSTEM_EVENTS);

/**
 * Estados que indican que el ticket está "activo" (no cerrado)
 */
export const ACTIVE_STATES = [
    TICKET_STATES.CREADO,
    TICKET_STATES.EN_ESPERA,
    TICKET_STATES.EN_PROCESO,
    TICKET_STATES.SOLUCIONADO,
    TICKET_STATES.REABIERTO
];

/**
 * Estados que permiten escalamiento
 */
export const ESCALATION_STATES = [
    TICKET_STATES.EN_ESPERA,
    TICKET_STATES.EN_PROCESO,
    TICKET_STATES.REABIERTO
];

File: src/api/constants/ticket_enums.py
"""
ticket_enums.py - Fuente única de verdad Backend
Enums para estados, eventos y campos del sistema de tickets
"""

from enum import Enum


class TicketState(str, Enum):
    """
    Estados del Ticket (formato BD con espacios)
    Backend usa espacios, Frontend usa guiones bajos
    """
    CREADO = 'creado'
    EN_ESPERA = 'en espera'
    EN_PROCESO = 'en proceso'
    SOLUCIONADO = 'solucionado'
    CERRADO = 'cerrado'
    REABIERTO = 'reabierto'
    
    @classmethod
    def values(cls):
        """Retorna lista de todos los valores"""
        return [e.value for e in cls]
    
    @classmethod
    def is_valid(cls, estado):
        """Verifica si un estado es válido"""
        return estado in cls.values()


class TicketEvent(str, Enum):
    """Eventos WebSocket del sistema"""
    CREATED = 'ticket_created'
    ASIGNADO = 'ticket_asignado'
    INICIADO = 'ticket_iniciado'
    SOLUCIONADO = 'ticket_solucionado'
    CERRADO = 'ticket_cerrado'
    REABIERTO = 'ticket_reabierto'
    ESCALADO = 'ticket_escalado'
    SOLICITUD_REAPERTURA = 'solicitud_reapertura'
    ELIMINADO = 'ticket_eliminado'
    ASIGNACION_ELIMINADA = 'asignacion_eliminada'
    
    @classmethod
    def values(cls):
        """Retorna lista de todos los valores"""
        return [e.value for e in cls]
    
    @classmethod
    def is_valid(cls, evento):
        """Verifica si un evento es válido"""
        return evento in cls.values()


class TicketFields(str, Enum):
    """Campos especiales del ticket"""
    SOLICITUD_PENDIENTE = 'tiene_solicitud_reapertura_pendiente'


# Mapeo: Evento → Estado resultante
EVENT_TO_STATE = {
    TicketEvent.CREATED: TicketState.CREADO,
    TicketEvent.ASIGNADO: TicketState.EN_ESPERA,
    TicketEvent.INICIADO: TicketState.EN_PROCESO,
    TicketEvent.SOLUCIONADO: TicketState.SOLUCIONADO,
    TicketEvent.CERRADO: TicketState.CERRADO,
    TicketEvent.REABIERTO: TicketState.REABIERTO,
    TicketEvent.ASIGNACION_ELIMINADA: TicketState.CREADO,
}

# Estados que permiten escalamiento
ESCALATION_STATES = [
    TicketState.EN_ESPERA,
    TicketState.EN_PROCESO,
    TicketState.REABIERTO
]

# Estados activos (no cerrados)
ACTIVE_STATES = [
    TicketState.CREADO,
    TicketState.EN_ESPERA,
    TicketState.EN_PROCESO,
    TicketState.SOLUCIONADO,
    TicketState.REABIERTO
]
